From 28c04b942cabcc97de14d53ccf6d60b6cc9561b0 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:05 -0400
Subject: [fs] btrfs: remove unnecessary filemap writting and waiting after block group relocation

Message-id: <1407179109-13714-59-git-send-email-sandeen@redhat.com>
Patchwork-id: 87356
O-Subject: [RHEL7.1 PATCH 058/422] Btrfs: remove unnecessary filemap writting and waiting after block group relocation
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 54eb72c05f7731b4b148da47419b90a5f2108036
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Fri Dec 13 18:30:44 2013 +0800

    Btrfs: remove unnecessary filemap writting and waiting after block group relocation

    We have commited transaction before, remove redundant filemap writting and
    waiting here, it can speed up balance relocation process.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/relocation.c b/fs/btrfs/relocation.c
index 63708f7..d8a82b8 100644
--- a/fs/btrfs/relocation.c
+++ b/fs/btrfs/relocation.c
@@ -4283,11 +4283,6 @@ int btrfs_relocate_block_group(struct btrfs_root *extent_root, u64 group_start)
   }
  }
 
- filemap_write_and_wait_range(fs_info->btree_inode->i_mapping,
-         rc->block_group->key.objectid,
-         rc->block_group->key.objectid +
-         rc->block_group->key.offset - 1);
-
  WARN_ON(rc->block_group->pinned > 0);
  WARN_ON(rc->block_group->reserved > 0);
  WARN_ON(btrfs_block_group_used(&rc->block_group->item) > 0);
-- 
1.7.1